smarts/core/default_map_builder.py

```diff
@@ -54,63 +97,41 @@ def get_road_map(map_spec) -> Tuple[RoadMap, str]:
     assert map_spec, "A road map spec must be specified"
     assert map_spec.source, "A road map source must be specified"
 
-    global _existing_map
-    if _existing_map:
-        if _existing_map.obj.is_same_map(map_spec):
-            return _existing_map.obj, _existing_map.map_hash
-        import gc
-
-        # Try to only keep one map around at a time...
-        del _existing_map
-        _existing_map = None
-        gc.collect()
-
-    if not os.path.isfile(map_spec.source):
-        scenario_root = Path(map_spec.source)
-        path_found = False
-        for i, map_name in enumerate(supported_maps):
-            map_paths = list(scenario_root.rglob(map_name))
-            if len(map_paths) == 1:
-                map_spec = replace(map_spec, source=str(map_paths[0]))
-                path_found = True
-                break
-            elif len(map_paths) > 1:
-                for map_path in map_paths:
-                    if not str(map_path).endswith("AUTOGEN.net.xml"):
-                        map_spec = replace(map_spec, source=str(map_path))
-                        path_found = True
-                        break
-                break
-            else:
-                continue
-
-        if not path_found:
-            raise FileNotFoundError(
-                f"Unable to find map in map_source={map_spec.source}."
-            )
-
-    road_map = None
-    if map_spec.source.endswith(".net.xml"):
-        # Keep this a conditional import so Sumo does not have to be
-        # imported if not necessary:
+    if os.path.isdir(map_spec.source):
+        map_type, map_source = _find_mapfile_in_dir(map_spec.source)
+    else:
+        map_type = _UNKNOWN_MAP
+        map_source = map_spec.source
+        if map_source.endswith(".net.xml"):
+            map_type = _SUMO_MAP
+        elif map_source.endswith(".xodr"):
+            map_type = _OPENDRIVE_MAP
+
+    if map_type == _SUMO_MAP:
         from smarts.core.sumo_road_network import SumoRoadNetwork
 
-        road_map = SumoRoadNetwork.from_spec(map_spec)
+        clazz = SumoRoadNetwork
```

Comment:
`clazz` lol ? Maybe you can use `map_class`.

Reply:
ok, done.